swap/swap.go

```diff
@@ -563,6 +629,16 @@ func (s *Swap) loadLastSentCheque(p enode.ID) (cheque *Cheque, err error) {
 	return cheque, err
 }
 
+// loadPendingCheque loads the current pending cheque for the peer from the store
+// and returns nil when there never was a pending cheque saved
+func (s *Swap) loadPendingCheque(p enode.ID) (cheque *Cheque, err error) {
+	err = s.store.Get(pendingChequeKey(p), &cheque)
+	if err == state.ErrNotFound {
+		return nil, nil
+	}
+	return cheque, err
```

Comment:
i don't like the practice of returning a non-nil `cheque` value if `err != nil && err != state.ErrNotFound`

i am aware this is also done by other `load` functions. 

thoughts?

Reply:
I changed it to always return `nil, err` on error in all load functions.